### PR TITLE
add: label_tex as an extention of ggplot2::label_parsed

### PR DESCRIPTION
**Pull request by atusy**
By this PR, math in facets' stripe becomes quite easy.

image

```r
library(dplyr)
library(ggplot2)
library(ggforce)
d <- data.frame(
    facet = "$f(x) = \\frac{1}{\\sqrt{2\\pi\\sigma^2}}exp\\left(-\\frac{(x-\\mu)^2}{2\\sigma^2}\\right)$",
    x = seq(-3, 3, 0.1)
  ) %>%
  mutate(y = dnorm(x))
ggplot(d, aes(x, y)) +
  geom_line() +
  facet_wrap(~ facet, labeller = label_tex)
```

### CODE REVIEW

**Comment by atusy** [comment]
I'm glad you like this. I force pushed with following changes

- moved latex2exp to suggests
- rebased to current `main` branch 
- coordinated with latest API of **ggplot2** and **latex2exp**

File: R/labeller.R [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+#' @inheritDotParams ggplot2::label_parsed -labels
+#'
+#' @examples
+#' library(ggplot2)
+#' d <- data.frame(x = 1, y = 1, facet = "$\\beta$")
+#' ggplot(d, aes(x, y)) +
+#'  geom_point() +
+#'  facet_wrap(~ facet, labeller = label_tex)
+#'
+#' @importFrom latex2exp TeX
```
Comment (thomasp85) [inline R/labeller.R:18]: This needs to be removed if in Suggest

File: R/labeller.R [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+#' @importFrom ggplot2 label_parsed
+#' @export
+label_tex <- function(labels, ...) {
+  label_parsed(
+    as.data.frame(lapply(labels, TeX, output = "character")),
```
Comment (thomasp85) [inline R/labeller.R:23]:
```suggestion
    as.data.frame(lapply(labels, latex2exp::TeX, output = "character")),
```

File: R/labeller.R [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+#' library(ggplot2)
+#' d <- data.frame(x = 1, y = 1, facet = "$\\beta$")
+#' ggplot(d, aes(x, y)) +
+#'  geom_point() +
+#'  facet_wrap(~ facet, labeller = label_tex)
+#'
+#' @importFrom latex2exp TeX
+#' @importFrom ggplot2 label_parsed
+#' @export
+label_tex <- function(labels, ...) {
```
Comment (thomasp85) [inline R/labeller.R:21]: You need to put a check for whether latex2exp is installed at the top of the function - something like 

```r
if (!requireNamespace("latex2exp", quiet = TRUE)) {
  stop("The latex2exp package is needed for this functionality", call. = FALSE)
}
```

**Review by thomasp85 — CHANGES_REQUESTED** [review]
Fantastic - just a few changes needed

File: R/labeller.R (hunk 1 above)
Comment (atusy) [reply R/labeller.R:18]: maybe we can wrap into unreachable if statement, so that we can still show the example code?

```
if (FALSE) {
  ...
}
```

Comment (thomasp85) [reply R/labeller.R:18]: it's the `@importFrom` - not the example

Comment (thomasp85) [reply R/labeller.R:18]: it's the `@importFrom` - not the example

Comment (atusy) [reply R/labeller.R:18]: oh, but I still have to remove here.

> @importFrom latex2exp TeX

Comment (thomasp85) [reply R/labeller.R:18]: yes - remove `@importFrom ...` - keep the example

**Comment by atusy** [comment]
Oops, I need to avoid string becoming factors by `as.data.frame` .

**Comment by thomasp85** [comment]
Once the last merge conflicts are fixed I can merge

**Comment by thomasp85** [comment]
thanks!!

**Comment by atusy** [comment]
Thank you, too!!!

**Pull request merged**